crates/uv-resolver/src/resolver/markers.rs

```diff
@@ -0,0 +1,30 @@
+use std::sync::Arc;
+
+use rustc_hash::FxHashMap;
+
+use uv_normalize::PackageName;
+use uv_pep508::MarkerTree;
+
+/// The superset of markers for which a package is known to be relevant.
+///
+/// These markers may not represent the exact set of relevant environments, as they aren't adjusted
+/// when backtracking; instead, we only _add_ to this set over the course of the resolution. As
+/// such, the marker value represents a superset of the environments in which the package is known
+/// to be included, but it may include environments in which the package is ultimately excluded.
+#[derive(Debug, Default, Clone)]
+pub(crate) struct KnownMarkers(Arc<FxHashMap<PackageName, MarkerTree>>);
+
+impl KnownMarkers {
+    /// Inserts the given [`MarkerTree`] for the given package name.
+    pub(crate) fn insert(&mut self, package_name: PackageName, marker_tree: MarkerTree) {
+        Arc::make_mut(&mut self.0)
```

Comment:
Is this a bit of a footgun? If you have two different `KnownMarkers` values and call `insert` on one of them, it will only mutate that one and not the other. It feels a little odd because they start by referencing the same object, and by virtue of mutation, they end up referring to different objects.

I haven't read on yet to see how this is used, but I'm guessing this is intended behavior.

Reply:
Hmmm it almost seems like you might not be benefiting much from the `Arc` here?

Feel free to ignore me if you've already considered all of this. :)